wpimath/src/main/java/edu/wpi/first/math/kinematics/Odometry.java

```diff
@@ -90,7 +90,7 @@ public Pose2d update(Rotation2d gyroAngle, T wheelPositions) {
 
     var newPose = m_poseMeters.exp(twist);
 
-    m_previousWheelPositions = m_kinematics.copy(wheelPositions);
+    m_previousWheelPositions = m_kinematics.copyInto(wheelPositions, m_previousWheelPositions);
```

Comment:
The return value of copyInto() seems redundant and can be removed.

Reply:
The problem is that if the wheel positions type is immutable, we can't modify `m_previousWheelPositions` in place.
Given that none of the standard WPILib ones are like that, though, maybe we could change to only supporting fully mutable wheel positions types?

Reply:
If we can't modify it in place, then how can copyInto() modify the buffer argument?

Reply:
It wouldn't modify the buffer argument, and would instead return the first argument (if it's completely and deeply immutable) or a new object. It's explained in the implementation note section of the doc comment:
```java
  /**
   <...>
   * <p>Implementation note:
   *
   * <ul>
   *   <li>If {@code P} is (deeply) immutable, then like {@link #copy}, this should return {@code
   *       positions}.
   *   <li>If {@code P} is completely mutable, then this should copy into {@code buffer} and return
   *       it.
   *   <li>Otherwise (some members are immutable and some are mutable), a new copy should be created
   *       and returned.
   * </ul>
   */
```

Reply:
Imo, we shouldn't add complexity for things we aren't using.